src/info/index_error.c

```diff
@@ -7,92 +7,110 @@
 #include "index_error.h"
 #include "rmalloc.h"
 #include "reply_macros.h"
+#include "util/timeout.h"
 
 extern RedisModuleCtx *RSDummyContext;
 
-const char* NA = "N/A";
-const char* IndexError_ObjectName = "Index Errors";
-const char* IndexingFailure_String = "indexing failures";
-const char* IndexingError_String = "last indexing error";
-const char* IndexingErrorKey_String = "last indexing error key";
+char* const NA = "N/A";
+char* const IndexError_ObjectName = "Index Errors";
+char* const IndexingFailure_String = "indexing failures";
+char* const IndexingError_String = "last indexing error";
+char* const IndexingErrorKey_String = "last indexing error key";
+char* const IndexingErrorTime_String = "last indexing error time";
+RedisModuleString* NA_rstr = NULL;
+
+static void initDefaultKey() {
+    NA_rstr = RedisModule_CreateString(RSDummyContext, NA, strlen(NA));
+    RedisModule_TrimStringAllocation(NA_rstr);
+}
```

Comment:
Any reason for this change, other than creating the `RedisModuleString` lazyly?

Reply:
I decided it would be better if we had a single constant RM string instead of allocating a new one for each error. Also by trimming it, we are sure we use as little memory as possible.